### DISCUSSION TITLE
Support `viewport-fit` custom metadata viewport property

### DISCUSSION THREAD

**Discussion by fywk** [vercel/next.js#46542] [Ideas]
### Describe the feature you'd like to request

With the new Metadata API on 13.2, you can customised the following properties for the viewport metadata:

```ts
export declare type Viewport = {
    width?: string | number;
    height?: string | number;
    initialScale?: number;
    minimumScale?: number;
    maximumScale?: number;
};
```

So apart from the options listed above, there's currently no other way to add or customise additional properties, such as `viewport-fit` for example.

The `viewport-fit=cover` property is particularly useful in allowing websites to span the whole screen, including the device’s sensor housing or notch area. Of course, additional CSS can be applied to add inset padding that respects the device's safe area. You could not, however, add this property with the current implementation.

### Describe the solution you'd like

Add `viewport-fit` as one of the available options that can be customised for the [meta viewport tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag).

### Describe alternatives you've considered

\-

**Comment by huozhi** [comment]
We've supported but the type was missing, this is fixed in 13.2.3

**Reply by pwrnrd** [reply]
What about `user-scalable`? Please see [metatag docs on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag) for an overview of all keys/types that should be supported by viewport.

**Reply by ciruz** [reply]
@pwrnrd hey, you can always set a string, like so:

```tsx
export const metadata = {
  title: "Home",
  description: "Welcome to Next.js",
  viewport:
    "width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0",
};
```

Result:

![image](https://user-images.githubusercontent.com/987947/222125129-b6b6758c-fd73-4774-8362-3be476b7006b.png)

**Reply by pwrnrd** [reply]
Thank you, that'll work. Perhaps from an API design it may be nice to either remove the object type Viewport or to include all Viewport items in the type. It seems a bit odd to me to have a type that only works in some cases.

**Reply by huozhi** [reply]
We've added user-scalable support in https://github.com/vercel/next.js/pull/46696

**Comment by y2hlbg** [comment]
we should add the below code into app/layout.tsx
```javascript
import type { Viewport } from "next";
export const viewport: Viewport = {
  // width: 'device-width',
  // initialScale: 1,
  // maximumScale: 1,
  // userScalable: false,
  viewportFit: 'cover',
}
```
